core/include/detray/core/mask_store.hpp

```diff
@@ -80,22 +100,35 @@ class mask_store {
      * @return vector of masks of a given type.
      */
     template <unsigned int mask_id>
-    constexpr const auto &group() const {
-        return std::get<mask_id>(_mask_tuple);
+    DETRAY_HOST_DEVICE constexpr const auto &group() const {
+        return detail::get<mask_id>(_mask_tuple);
     }
 
     /** Access underlying container - const
      *
      * @return internal masks tuple type
      */
+    DETRAY_HOST_DEVICE
     const auto &masks() const { return _mask_tuple; }
 
     /** Access underlying container
      *
      * @return internal masks tuple type
      */
+    DETRAY_HOST_DEVICE
     const auto &masks() { return _mask_tuple; }
 
+    /** Obtain the vecmem data of mask store
+     *
+     * @return tuple type of vecmem::data::vector_view objects
+     */
+    template <std::size_t... ints>
+    DETRAY_HOST tuple_type<vecmem::data::vector_view<mask_types>...> data(
+        std::index_sequence<ints...> /*seq*/) {
+        return thrust::make_tuple(vecmem::data::vector_view<mask_types>(
+            vecmem::get_data(detail::get<ints>(_mask_tuple)))...);
+    }
```

Comment:
We are passing ```vecmem::memory_resource``` to all constructors everywhere currently, which is why I gave up on using other vector types for now. I wasn't really a fan of this, but I was told that it is better to build the memory resource at a global scope. I was planning to have a look at class constructors.

Long term, I was planning to have a class (or more) that does this kind of memory setup, also because I need different serialization of the data in memory for horizontal vectorization on host side. The other classes should only be concerned of getting the indexing set up correctly and provide an access interface (in my mind)

Reply:
fixed it - removed data() member function and only use standalone function